### PR TITLE
Add glossary page

### LINKED ISSUE #92: Add a glossary of terms
Those managing coding projects or new to coding may find a glossary useful. This should explain the common terms and tools in simplified language.

### PR DESCRIPTION
**Pull request by foster999**
Adds a glossary of terms page as a very high level introduction to terms around coding, analysis tools and practices

Closes #92

### CODE REVIEW

File: book/glossary.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+### Continuous integration and continuous delivery (CI/CD)
+
+CI/CD tools help to automate parts of your code development workflow. This can include running automated tests against your code and deploying the latest version of your code into production. GitHub Actions is an example tool for running CI/CD workflows.
+
+Continuous integration describes regularly combining code changes from multiple contributors on a single software project. Integrating these changes regularly helps to check that independent changes work correctly together. Continous delivery describes automation of the software release process. Automating both of these using a CI/CD tool increases the efficiency of building software and getting it into production.
+
+### Virtual environment
+
+Virtual environments are tools for managing dependencies and isolating projects. A virtual environment is effectively a folder tree containing the programming language version and a list of dependencies.
```
Comment (foster999) [inline book/glossary.md]: Could add that this helps us to keep track of dependencies and avoid conflicts between different projects?

File: book/glossary.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+Continuous integration describes regularly combining code changes from multiple contributors on a single software project. Integrating these changes regularly helps to check that independent changes work correctly together. Continous delivery describes automation of the software release process. Automating both of these using a CI/CD tool increases the efficiency of building software and getting it into production.
+
+### Virtual environment
+
+Virtual environments are tools for managing dependencies and isolating projects. A virtual environment is effectively a folder tree containing the programming language version and a list of dependencies. 
+
+### Integrated development environment (IDE)
+
+An IDE is a tool that facilitates software development for a given progamming language. An IDE will usually consist of a code editor and debugger.
```
Comment (foster999) [inline book/glossary.md]: Could also mention development automation tools, for example auto-complete.

File: book/glossary.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+### Code interpreter
+
+A computer program that runs code in a particular programming language. For example, the program that reads your Python or R analysis code and runs it. A non-interactive interpreter runs code in order, which is important for reproducibility.
+
+Interactive interpreters allow you to run individual lines of code, which means that code can be run out of order. Notebooks use interactive interpreters. These are not suitable for running analysis pipelines, because they do not ensure that the code is run reproducibly.
+
+### Debugger
+
+A computer program that is used to test software and to identify the root cause of errors or "bugs". Debuggers allow you to pause the code at specific points and to  walk through code step by step, in order to understand how it is working.
```
Comment (foster999) [inline book/glossary.md]:
```suggestion
A computer program that is used to test software and to identify the root cause of errors or "bugs". Debuggers allow you to pause the code at specific points and to walk through code step by step, in order to understand how it is working.
```

**Review by joshlynchONS — APPROVED** [review]
This looks good for a first draft.

**Pull request merged**